### ISSUE TITLE
parametrize CallbackLogic in CallbackController

### ISSUE THREAD

**Issue by unkarjedy** [pac4j/play-pac4j#243]
For some reason org.pac4j.play.CallbackController has a hardcoded CallbackLogic:
```java 
private CallbackLogic<Result, PlayWebContext> callbackLogic = new DefaultCallbackLogic<>();
```
Callback logic can be set in `org.pac4j.core.config.Config`:
```java
public void setCallbackLogic(final CallbackLogic callbackLogic)
```
but it is not used.

It would be great to be able to parametrize the `callbackLogic`, parhaps to read it from config after it is injected.

**Comment by unkarjedy** [comment]
probably related with https://github.com/pac4j/play-pac4j/issues/242 because config is injected

**Comment by leleuj** [comment]
Generally, you only have one `CallbackController` and thus one `CallbackLogic`. This is a bit strange to be able to set the `CallbackLogic` at the `Config` logic, I honestly don't remember the reason behind this design.
Do you really have a use case for that?

**Comment by unkarjedy** [comment]
@leleuj The only reason we have only one `CallbackController` is just that it delegates the real work to `CallbackLogic`. There is no need for extra abstraction `interface CallbackLogic` while we do not allow user to change it. The main purpose of `Strategy Pattern` is to be able to change the logic at a runtime. 

In my current prod project, I've created a custom `MyCallbackLogic extends DefaultCallbackLogic[Result, PlayWebContext]` that a bit modifies `redirectToOriginallyRequestedUrl` (to support redirection to SinglePageApplications that are deployed on a separate domain). And just because `CallbackController` has a hardcoded logic I had to create my custom `MyCallbackController`. So I create 2 classes instead of 1. More of that I have to extend Java-based `play.mvc.Controller` in my Scala code just because `PlayWebContext` requires Java-based `play.mvc.Context`

Actually the same logic for `LogoutController` and `DefaultLogoutLogic`

If you agree that we should be able to change the default logic **I could prepare a pull-request**.
